src/prettyprint.ts

```diff
@@ -130,6 +130,10 @@ export function format(src: string, indentation: number = 4, useSpaces: boolean
     }
 
     htmlResult.rootNodes.forEach(node => {
+        const doctype = node.sourceSpan.start.getContext(20, 1).before;
```

Comment:
I see two problems with this solution:

1) If the cdata node is longer than 20 characters then this misses it.

2) If you have a comment as a root node which is less than 20 characters then each time the formatter runs, it will duplicate the comment.

We're using the Angular html parser in this extension, and unfortunately it drops the CData tokens while parsing (https://github.com/angular/angular/blob/master/packages/compiler/src/ml_parser/parser.ts#L105). The ideal solution would be to use an html parser that doesn't remove nodes.

We could check to see if a file begins with `<!DOCTYPE` and then include everything up to the first `>` character. Something like

```typescript
if(src.trim().startsWith('<!DOCTYPE')) {
    const trimmed = src.trim();
    pretty.push(trimmed.substr(0, trimmed.indexOf('>')+1));
}
```

It isn't a comprehensive fix, it only works if the `<!DOCTYPE` node is at the beginning of the file. But it may be good enough.